oak/server/oak_grpc_node.cc

```diff
@@ -53,26 +56,33 @@ std::unique_ptr<OakGrpcNode> OakGrpcNode::Create() {
   return node;
 }
 
-grpc::Status OakGrpcNode::Start() {
+void OakGrpcNode::Start() {
   // Start a new thread to process the gRPC completion queue.
   std::thread thread(&OakGrpcNode::CompletionQueueLoop, this);
   thread.detach();
   // TODO: This thread will need to be joined once we sort out termination.
-  return grpc::Status::OK;
 }
 
 void OakGrpcNode::CompletionQueueLoop() {
   LOG(INFO) << "Starting gRPC completion queue loop";
+
+  Handle req_handle = FindChannel(kGrpcNodeRequestPortName);
+  Handle rsp_handle = FindChannel(kGrpcNodeResponsePortName);
+  ChannelHalf* req_half = BorrowChannel(req_handle);
+  ChannelHalf* rsp_half = BorrowChannel(rsp_handle);
+
   // The stream object will delete itself when finished with the request,
   // after creating a new stream object for the next request.
-  auto stream = new ModuleInvocation(module_service_.get(), completion_queue_.get(),
-                                     req_half_.get(), rsp_half_.get());
+  auto stream =
+      new ModuleInvocation(module_service_.get(), completion_queue_.get(), req_half, rsp_half);
   stream->Start();
   while (true) {
     bool ok;
     void* tag;
     if (!completion_queue_->Next(&tag, &ok)) {
-      LOG(FATAL) << "Failure reading from completion queue";
+      if (!termination_pending_.load()) {
```

Comment:
Should this be moved up to just inside the while loop?

Reply:
I don't think so &ndash; the loop will always terminate because of a `Next()` failure, and the flag is just used so we can distinguish between normal shutdown and something abnormal.